src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -33,38 +31,39 @@
     public CartDto createCart(){
         return mapper.mapToCartDto(service.createCart());
     }
+
     @GetMapping("/{cartId}")
-    public CartDto getCart(@PathVariable Long cartId) throws CartNotFoundException {
+    public CartDto getCart(@PathVariable Long cartId) {
         return mapper.mapToCartDto(service.getCart(cartId));
-    };
+    }
 
     @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
-    public CartDto updateCart(@RequestBody Cart cart) throws CartNotFoundException {
-        return mapper.mapToCartDto(service.saveCart(service.getCart(cart.getId())));
-    };
+    public CartDto updateCart(@RequestBody CartDto cartDto) {
+        return mapper.mapToCartDto(service.updateCart(mapper.mapToCart(cartDto)));
```

Comment:
Zła metoda -> service.updateCart u Ciebie to jakaś staroć, ta właściwa metoda w Twoim serwisie nazywa się saveCart. Albo tu zrób saveCart, albo w serwisie saveCart przemianuj na updateCart, a stary updateCart po prostu wykasuj (jest niżej pod getItem).
Poza tym saveCart robi pewną niebezpieczną rzecz -> jeśli id, którego chcesz zrobić update (w zasadzie my tu robimy update tylko do jednego pola - closed), nie istnieje, to zamiast wyrzucić błąd o braku takiego id, zapisuje zupełnie nowy Cart .... Brakuje sprawdzenia, czy taki id istnieje,